File: 프로그래머스/Lv2/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer(2,0);
    map<string,int> m;
    int Index=1;
    m[words[0]]++;
    int people=1;
    int tern=1;
    
    while(Index<words.size()){
        people++;
        people=people%n;
        m[words[Index]]++;
        if(m[words[Index]]>=2 || words[Index-1].back()!=words[Index].front()){
            if(people==0) people=n;
            answer.front()=people;
            answer.back()=tern;
            break;
        }
        
        Index++;
        if(Index%n==0) tern++;
    }

    return answer;
}

File: 프로그래머스/Lv2/영어 끝말잇기.py
def solution(n, words):
    answer = [0,0]
    s={words[0]}
    last=words[0][-1]
    
    for i in range(1,len(words)):
        if(words[i][0]!=last or (words[i] in s)):
            
            i+=1
            cycle = int(i/n) #몇바퀴
            if(i%n==0): answer=[n,cycle]
            else: answer=[i-n*cycle,cycle+1]
                
            break
        else:
            s.add(words[i])
            last=words[i][-1]

    return answer
